src/item.h

```diff
@@ -704,8 +704,9 @@ class item : public JsonSerializer, public JsonDeserializer
         /**
          * Callback when a player starts wielding the item. The item is already in the weapon
          * slot and is called from there.
+         * @param mv number of moves *already* spent wielding the weapon
          */
-        void on_wield( player &p );
+        void on_wield( player &p, int mv = 0 );
```

Comment:
This `mv` parameter looks very un-intuitive (especially the "*already* spent" part). But I don't see any better solution than moving all the message-generating code into `player::wield`, which might not be a better solution at all.

Btw. the code here looks quite good.